### PR TITLE
osdc: Change 'bool budgeted' to 'int budget' to avoid recalculating.

### PR DESCRIPTION
**Pull request by majianpeng**
Avoing recalculating in op finish stage.

Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

### CODE REVIEW

**Comment by majianpeng** [comment]
@gregsfortytwo . please review this. Thanks

**Comment by batrick** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
i suspect that this change led to the timeouts in my test batch.

**Comment by majianpeng** [comment]
@tchaikov . How to reproduce  the timeout problem?

**Comment by tchaikov** [comment]
@majianpeng i will run your change thru rados qa suite again to see if it's who failed the test.

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -3139,8 +3139,10 @@ void Objecter::_finish_op(Op *op, int r)
 
   // op->session->lock is locked unique or op->session is null
 
-  if (!op->ctx_budgeted && op->budgeted)
-    put_op_budget(op);
+  if (!op->ctx_budgeted && op->budget) {
```
Comment (tchaikov) [inline src/osdc/Objecter.cc]: what if the op has no budget? in that case, we should still `op_throttle_ops.put(1);`, shouldn't we?

File: src/osdc/Objecter.cc [hunk 2]
```diff
@@ -3140,8 +3140,10 @@ void Objecter::_finish_op(Op *op, int r)
 
   // op->session->lock is locked unique or op->session is null
 
-  if (!op->ctx_budgeted && op->budgeted)
-    put_op_budget(op);
+  if (!op->ctx_budgeted && op->budget >= 0) {
```
Comment (tchaikov) [inline src/osdc/Objecter.cc]: no, this is still buggy. again, what if the op has no budget? in that case, the op->budget is -1, and "op_throttle_ops.put(1);" is not called in that case. op_throttle_ops will not reduced after the op w/o budget completes.

Comment (majianpeng) [reply src/osdc/Objecter.cc]: calc_op_budget return value >=0.  Only budget >=0, it call op_throttle_ops.put(1).

**Comment by majianpeng** [comment]
Thanks! @tchaikov

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-18_11:46:34-rados-wip-kefu-testing-2018-04-18-1811-distro-basic-mira/

**Comment by majianpeng** [comment]
@tchaikov . i can't find the reason which cause tests dead. If omit put, this mean budget < 0. If budget < 0, the get_or_fail will assert.  Have you good  suggestions?

**Comment by tchaikov** [comment]
@majianpeng i don't think the test failure is relevant. 

could you note down what you change in the latest version of your PR? i noticed that you pushed a new version after i ran it thru rados qa suite.

**Comment by majianpeng** [comment]
Only rebase on master. Don't change any code.

**Pull request merged**